From 4a6e293c5fbd04e6fd322823846c3df01b3abeb1 Mon Sep 17 00:00:00 2001
From: Jaikrishna Nemallapudi <jaikrishnax.nemallapudi@intel.com>
Date: Wed, 22 Mar 2017 19:16:23 +0530
Subject: [PATCH 4091/4301] ASoC: Intel: Fix Compilation issues for probe
 compress APIs

commit 92997222d96ed09d5fb17e3c22de27ed6964389c from
git://git@github.com:01org/linux-intel-4.9.git

Select the SND_SOC_COMPRESS and move the inline to header file.

Fixes: 202dcc0("ASoC: Intel: Add Probe compress APIs")
Signed-off-by: Jaikrishna Nemallapudi <jaikrishnax.nemallapudi@intel.com>
Cc: Divya Prakash <divya1.prakash@intel.com>
Cc: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/Kconfig                      |    1 +
 sound/soc/intel/skylake/skl-compress.c |    6 +-----
 sound/soc/intel/skylake/skl-compress.h |   10 ++++++++--
 3 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/sound/soc/Kconfig b/sound/soc/Kconfig
index 182d92e..180a930 100644
--- a/sound/soc/Kconfig
+++ b/sound/soc/Kconfig
@@ -9,6 +9,7 @@ menuconfig SND_SOC
 	select SND_JACK
 	select REGMAP_I2C if I2C
 	select REGMAP_SPI if SPI_MASTER
+	select SND_SOC_COMPRESS
 	---help---
 
 	  If you want ASoC support, you should say Y here and also to the
diff --git a/sound/soc/intel/skylake/skl-compress.c b/sound/soc/intel/skylake/skl-compress.c
index c8b26e8..de58eb8 100644
--- a/sound/soc/intel/skylake/skl-compress.c
+++ b/sound/soc/intel/skylake/skl-compress.c
@@ -25,11 +25,7 @@
 #include <sound/pcm_params.h>
 #include <sound/soc.h>
 #include "skl.h"
-inline
-struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream)
-{
-	return stream->runtime->private_data;
-}
+#include "skl-compress.h"
 
 struct hdac_ext_bus *get_bus_compr_ctx(struct snd_compr_stream *substream)
 {
diff --git a/sound/soc/intel/skylake/skl-compress.h b/sound/soc/intel/skylake/skl-compress.h
index 9fcf6c38f..888ef6f 100644
--- a/sound/soc/intel/skylake/skl-compress.h
+++ b/sound/soc/intel/skylake/skl-compress.h
@@ -19,8 +19,8 @@
  *
  */
 
-inline
-struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream);
+#ifndef __SOUND_SOC_SKL_COMPRESS_H
+#define __SOUND_SOC_SKL_COMPRESS_H
 struct hdac_ext_bus *get_bus_compr_ctx(struct snd_compr_stream *substream);
 void skl_set_compr_runtime_buffer(struct snd_compr_stream *substream,
 				struct snd_dma_buffer *bufp, size_t size);
@@ -32,4 +32,10 @@ int skl_substream_alloc_compr_pages(struct hdac_ext_bus *ebus,
 int skl_compr_free_pages(struct snd_compr_stream *substream);
 int skl_substream_free_compr_pages(struct hdac_bus *bus,
 				struct snd_compr_stream *substream);
+static inline
+struct hdac_ext_stream *get_hdac_ext_compr_stream(struct snd_compr_stream *stream)
+{
+	return stream->runtime->private_data;
+}
 
+#endif /* __SOUND_SOC_SKL_COMPRESS_H */
-- 
1.7.5.4

